File: vaultx-backend/src/main/java/com/vaultx/service/CollectionService.java
package com.vaultx.service;

import com.vaultx.dto.document.CollectionDto;
import java.util.List;
import java.util.UUID;

public interface CollectionService {
    List<CollectionDto> getUserCollections();
    CollectionDto getCollection(UUID id);
    CollectionDto createCollection(String name, String description);
    CollectionDto renameCollection(UUID id, String name, String description);
    void addDocumentsToCollection(UUID collectionId, List<UUID> documentIds);
    void removeDocumentFromCollection(UUID collectionId, UUID documentId);
    void deleteCollection(UUID id);
    byte[] downloadCollection(UUID id);
}

File: vaultx-frontend/src/services/collectionService.js
import api from './api';

const collectionService = {
  getCollections: () => api.get('/collections'),
  
  getCollection: (id) => api.get(`/collections/${id}`),
  
  createCollection: (name, description) =>
    api.post('/collections', { name, description }),
  
  renameCollection: (id, name, description) =>
    api.put(`/collections/${id}`, { name, description }),
  
  addDocuments: (id, documentIds) =>
    api.post(`/collections/${id}/documents`, { documentIds }),
  
  removeDocument: (collectionId, documentId) =>
    api.delete(`/collections/${collectionId}/documents/${documentId}`),
  
  deleteCollection: (id) => api.delete(`/collections/${id}`),

  downloadCollection: (id) => api.get(`/collections/${id}/download`, { responseType: 'blob' })
};

export default collectionService;

